inference-engine/thirdparty/clDNN/runtime/ocl/ocl_memory.cpp

```diff
@@ -306,11 +309,17 @@ gpu_usm::gpu_usm(ocl_engine* engine, const layout& layout, allocation_type type)
 }
 
 void* gpu_usm::lock(const stream& stream, mem_lock_type /*type*/) {
-    assert(get_allocation_type() != allocation_type::usm_device && "Can't lock usm device memory!");
```

Comment:
* Copy memory only when mem_lock_type is `read`.
* If mem_lock_type is `write` and allocation type is `usm_device`, it should throw exception.

Reply:
Agree, add exception code in case NOT read type.